clients/venice-thin-client/src/main/java/com/linkedin/venice/client/store/AbstractAvroStoreClient.java

```diff
@@ -884,6 +926,17 @@ public void onHeaderReceived(Map<String, String> headers) {
       if (schemaIdHeader != null) {
         responseSchemaId = Integer.parseInt(schemaIdHeader);
       }
+      String responseTypeHeader = headers.get(HttpConstants.VENICE_RESPONSE_TYPE);
+      if (responseTypeHeader != null) {
+        responseType = Integer.parseInt(responseTypeHeader);
+      }
+      if (responseFormatProvider != null) {
+        ResponseFormat format = responseFormatProvider.apply(RequestType.valueOf(responseType));
+        this.envelopeDeserializerFunc = format.envelopeDeserializerFunc();
+        this.recordDeserializerFunc = format.recordDeserializerFunc();
+        this.indexRetrievalFunc = format.indexRetrievalFunc();
+        this.streamingFooterRecordDeserializer = format.streamingFooterRecordDeserializer();
+      }
```

Comment:
This piece seems to belong the constructor function.
Do you think `responseFormatProvider` could change after the initialization?